ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -0,0 +1,274 @@
+        else
+          let _, enums = string_of_ty_with_enums t in
+          enums
+
+  let enums_from_params ps =
+    List.fold_left
+      (fun enums p ->
+        let _t, e = string_of_ty_with_enums p.param_type in
+        StringMap.union choose_enum enums e
+      )
+      StringMap.empty ps
+
+  let session_id =
+    {
+      param_type= Ref Datamodel_common._session
+    ; param_name= "session_id"
+    ; param_doc= "Reference to a valid session"
+    ; param_release= Datamodel_common.rio_release
+    ; param_default= None
+    }
+
+  let enums_in_messages_of_obj obj =
+    List.fold_left
+      (fun enums msg ->
+        let params =
+          if msg.msg_session then
+            session_id :: msg.msg_params
+          else
+            msg.msg_params
+        in
+        let enums1 = enums_from_result obj msg in
+        let enums2 = enums_from_params params in
+        enums
+        |> StringMap.union choose_enum enums1
+        |> StringMap.union choose_enum enums2
+      )
+      StringMap.empty obj.messages
+
+  let get_modules messages has_time_type =
+    match messages with
+    | [] ->
+        `Null
+    | _ ->
+        let items =
+          match has_time_type with
+          | true ->
+              [
+                `O [("name", `String "fmt"); ("sname", `Null)]
+              ; `O [("name", `String "time"); ("sname", `Null)]
+              ]
+          | false ->
+              [`O [("name", `String "fmt"); ("sname", `Null)]]
+        in
+        `O [("import", `Bool true); ("items", `A items)]
+
+  let xenapi objs =
+    let enums_acc = ref StringMap.empty in
+    let erase_existed enums =
+      let enums =
+        StringMap.filter (fun k _ -> not (StringMap.mem k !enums_acc)) enums
+      in
+      enums_acc := StringMap.union choose_enum !enums_acc enums ;
+      enums
+    in
+    List.map
+      (fun obj ->
+        let fields, enums_in_fields, has_time_type =
+          fields_of_obj_with_enums obj
+        in
+        let enums_in_msgs = enums_in_messages_of_obj obj in
+        let enums =
+          let enums =
+            enums_in_fields
+            |> StringMap.union choose_enum enums_in_msgs
+            |> erase_existed
+          in
+          StringMap.fold (fun k v acc -> of_enum k v :: acc) enums []
+        in
+        let event_snapshot =
+          if String.lowercase_ascii obj.name = "event" then
+            [
+              `O
+                [
+                  ("name", `String "Snapshot")
+                ; ( "description"
+                  , `String
+                      "The record of the database object that was added, \
+                       changed or deleted"
+                  )
+                ; ("type", `String "RecordInterface")
+                ]
+            ]
+          else
+            []
+        in
+        let obj_name = snake_to_camel obj.name in
+        let modules = get_modules obj.messages has_time_type in
+        let event_session_value = function
```

Comment:
I think `event_session_value` and `event_snapshot` can be moved into `CommonFunctions.ml`.

Reply:
Ah. I see. They are go-specific things. Please ignore this comment. Sorry for this.